esti/lakectl_local_test.go

```diff
@@ -484,105 +477,189 @@ func TestLakectlLocal_interruptedPull(t *testing.T) {
 	runCmd(t, Lakectl()+" log lakefs://"+repoName+"/"+mainBranch, false, false, vars)
 
+	prefix := "images"
+
 	tests := []struct {
 		name   string
 		prefix string
 	}{
 		{
-			name:   "root",
-			prefix: "",
-		},
-		{
-			name:   "prefix",
-			prefix: "images",
+			name:   prefix,
+			prefix: prefix,
 		},
 	}
 	for _, tt := range tests {
 		t.Run(tt.name, func(t *testing.T) {
 			dataDir, err := os.MkdirTemp(tmpDir, "")
 			require.NoError(t, err)
-			vars["PREFIX"] = "/" + tt.prefix
+
+			runCmd(t, Lakectl()+" branch create lakefs://"+repoName+"/"+tt.name+" --source lakefs://"+repoName+"/"+mainBranch, false, false, vars)
+
 			vars["LOCAL_DIR"] = dataDir
+			vars["PREFIX"] = ""
 			vars["BRANCH"] = tt.name
 			vars["REF"] = tt.name
-			runCmd(t, Lakectl()+" branch create lakefs://"+repoName+"/"+vars["BRANCH"]+" --source lakefs://"+repoName+"/"+mainBranch, false, false, vars)
-			RunCmdAndVerifyContainsText(t, Lakectl()+" local clone lakefs://"+repoName+"/"+vars["BRANCH"]+vars["PREFIX"]+" "+dataDir, false, "Successfully cloned lakefs://${REPO}/${REF}${PREFIX} to ${LOCAL_DIR}.", vars)
+			RunCmdAndVerifyContainsText(t, Lakectl()+" local clone lakefs://"+repoName+"/"+vars["BRANCH"]+"/"+vars["PREFIX"]+" --pre-sign=false "+dataDir, false, "Successfully cloned lakefs://${REPO}/${REF}/${PREFIX} to ${LOCAL_DIR}.", vars)
+
+			idx := local.Index{}
+			indexDir := os.DirFS(dataDir)
+			yamlFile, err := fs.ReadFile(indexDir, ".lakefs_ref.yaml")
 
-			// Upload and commit a large object, so we will have a chance to interrupt the pull before it finishes
-			fileName := "test.txt"
-			fd, err := os.Create(fileName)
 			require.NoError(t, err)
-			require.NoError(t, fd.Truncate(1e8))
-			require.NoError(t, fd.Close())
-			runCmd(t, Lakectl()+" fs upload -s "+fileName+" lakefs://"+repoName+"/"+vars["BRANCH"]+vars["PREFIX"]+"/"+fileName, false, false, vars)
-			runCmd(t, Lakectl()+" commit lakefs://"+repoName+"/"+vars["BRANCH"]+" --allow-empty-message -m \" \"", false, false, vars)
 
-			// Pull changes and interrupt
-			RunCmdAndVerifyContainsTextWithTimeout(t, Lakectl()+" local pull "+dataDir, true, false, "", vars, time.Millisecond*100)
+			err = yaml.Unmarshal(yamlFile, &idx)
+			require.NoError(t, err)
 
-			// Pull changes without force flag
-			runCmd(t, Lakectl()+" local pull "+dataDir, true, false, vars)
+			idx.ActiveOperation = "pull"
+			updatedYAML, err := yaml.Marshal(&idx)
+			require.NoError(t, err)
 
-			// Pull changes and verify data
-			runCmd(t, Lakectl()+" local pull "+dataDir+" --force", false, false, vars)
-			localVerifyDirContents(t, dataDir, []string{fileName})
+			err = os.WriteFile(dataDir+"/.lakefs_ref.yaml", updatedYAML, 0644)
+			require.NoError(t, err)
+
+			// Pull without force flag
+			expectedRaw := `Latest pull operation was interrupted, local data may be incomplete.
+Use "lakectl local pull... --force" to sync with the remote.`
+			sanitizedResult := runCmd(t, Lakectl()+" local pull "+dataDir, true, false, vars)
```

Comment:
Why can't we use RunCmdAndVerifyContainsText?

Reply:
Because it doesn't have a `expectFail` flag, it assume it shouldn't fail